locust/test/test_dispatch.py

```diff
@@ -3591,6 +3591,85 @@ class User5(User):
                                 )
 
 
+class TestWeight(unittest.TestCase):
+    def test_set_weight_outside_class(self):
+        class User1(User):
+            pass
+
+        class User2(User):
+            pass
+
+        User1.weight = 0
+
+        worker_node1 = WorkerNode("1")
+        worker_node2 = WorkerNode("2")
+        users_dispatcher = UsersDispatcher(worker_nodes=[worker_node1, worker_node2], user_classes=[User1, User2])
+
+        users_dispatcher.new_dispatch(target_user_count=30, spawn_rate=0.5)
+        users_dispatcher._wait_between_dispatch = 0
+        iterations = list(users_dispatcher)
+
+        # Here returns: {"User1": 0, "User2": 15}
+        self.assertDictEqual(iterations[-1]["1"], {"User1": 0, "User2": 30}) 
+
+    def test_set_weight_inside_class(self):
+        class User1(User):
+            weight = 0
+
+        class User2(User):
+            pass
+
+        class User3(User):
+            weight = 0
+
+        worker_node1 = WorkerNode("1")
+        worker_node2 = WorkerNode("2")
+        worker_node3 = WorkerNode("3")
+        users_dispatcher = UsersDispatcher(worker_nodes=[worker_node1, worker_node2, worker_node3], user_classes=[User1, User2, User3])
+
+        users_dispatcher.new_dispatch(target_user_count=30, spawn_rate=0.5)
+        users_dispatcher._wait_between_dispatch = 0
+        iterations = list(users_dispatcher)
+
+        # Here returns: {"User1": 0, "User2": 10, "User3": 0}
+        self.assertDictEqual(iterations[-1]["1"], {"User1": 0, "User2": 30, "User3": 0}) 
```

Comment:
Looks like a bug where it's deciding how many users to add based on the number of classes rather than the weights those classes have